Refactor FXIOS-7301 [Swiftlint] Remove 1 closure_body_length violation from RustAutofill.swift and decrease threshold

ionixjunior opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7301)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16442)

## :bulb: Description
This PR removes 1 `closure_body_length` violation from the `RustAutofill.swift` file. Also, this PR decrease the warning and error threshold to 50.

Unlike previous pull requests, this one addresses some switch-case statements that aren't easy to define. I'm proposing in this PR to name the new functions based on the comments contained in each block of code.

This PR is part of a series of PRs described here https://github.com/mozilla-mobile/firefox-ios/issues/16442#issuecomment-2197676804 and here https://github.com/mozilla-mobile/firefox-ios/issues/16442#issuecomment-2549957143.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

FilippoZazzeroni commented on firefox-ios/Storage/Rust/RustAutofill.swift:
```diff
@@ -512,6 +464,74 @@ public class RustAutofill {
+        self.logger.log("Autofill key lost, new one generated",
+                        level: .warning,
+                        category: .storage)
+        self.resetCreditCardsAndKey(rustKeys: rustKeys, completion: completion)
```
Nit: self is not needed here

FilippoZazzeroni commented on firefox-ios/Storage/Rust/RustAutofill.swift:
```diff
@@ -512,6 +464,74 @@ public class RustAutofill {
+        // The key is present, but we didn't expect it to be there.
+        // or
+        // We expected the key to be present, but it's gone missing on us
+        self.logger.log("Autofill key lost, new one generated",
```
same

FilippoZazzeroni reviewed (commented):
Nit most of the `self` can be removed from the sub methods. The rest great work 🎉 🎉

ionixjunior commented:
Thank you for catching that, @FilippoZazzeroni! I've updated the branch and rebased to incorporate your feedback. Let me know if you need anything else.

FilippoZazzeroni reviewed (approved):
Perfect @ionixjunior great work 🎉 , approving.
